packages/react/src/VideoPlayer/VideoPlayer.tsx

```diff
@@ -0,0 +1,188 @@
+import React, {useState, useRef, useEffect} from 'react'
+import clsx from 'clsx'
+import type {BaseProps} from '../component-helpers'
+import {Text} from '../Text'
+import {Controls} from './components'
+import {MarkGithubIcon} from '@primer/octicons-react'
+
+/**
+ * Design tokens
+ */
+import '@primer/brand-primitives/lib/design-tokens/css/tokens/functional/components/video-player/base.css'
+
+/** * Main Stylesheet (as a CSS Module) */
+import styles from './VideoPlayer.module.css'
+
+type VideoPlayerProps = {
+  poster?: string
+  title: string
+  branding?: boolean
+  children: React.ReactElement | React.ReactElement[]
+  ref?: React.RefObject<HTMLVideoElement>
+} & React.HTMLProps<HTMLVideoElement> &
+  BaseProps<HTMLVideoElement>
+
+const Root = ({
+  poster,
+  title,
+  branding = true,
+  children,
+  className,
+  onPlay,
+  onPause,
+  onLoadedMetadata,
+  onPlaying,
+  ref,
+  ...rest
+}: VideoPlayerProps) => {
+  const videoPlayerRef = useRef<HTMLVideoElement>(null)
+  const videoWrapperRef = useRef<HTMLDivElement>(null)
+  const videoRef = ref ? ref : videoPlayerRef
+  const [playing, setPlaying] = useState(false)
+  const [playedTime, setPlayedTime] = useState(0)
+  const [totalTime, setTotalTime] = useState(0)
+  const [trackInformation, setTrackInformation] = useState<TextTrackCueList | undefined>(undefined)
+
+  /* > Play / Pause                                                    */
+  const handleVideoPlayback = () => {
+    if (videoRef.current) {
+      if (videoRef.current.paused) {
+        videoRef.current.play()
+      } else {
+        videoRef.current.pause()
+      }
+    }
+  }
+
+  const handleVideoOnPlaying = () => {
+    videoRef.current?.textTracks[0].cues && setTrackInformation(videoRef.current.textTracks[0].cues)
+  }
+
+  /* > Hide Default Captions                                                   */
+  useEffect(() => {
+    if (videoRef.current) {
+      for (let i = 0; i < videoRef.current.textTracks.length; i++) {
+        videoRef.current.textTracks[0].mode = 'hidden'
+      }
+    }
+  }, [videoRef])
+
+  /* > Set breakpoint classes                                                   */
+
+  useEffect(() => {
+    const handleResize = () => {
+      const breakpoint = videoWrapperRef.current?.getBoundingClientRect().width
+      if (breakpoint && breakpoint < 650) {
+        videoWrapperRef.current.classList.add(styles.VideoPlayer__breakpointSmall)
+      } else {
+        videoWrapperRef.current?.classList.remove(styles.VideoPlayer__breakpointSmall)
+      }
+    }
```

Comment:
I'm likely missing something crucial here, but AFAICT this is just adding some classes for visual styling. If that's the case, can this not be done with css media queries instead of in the JS runtime?

Reply:
I was handling this specifically based on the size of the container which I would hope to replace with container queries that are supported but not yet available in our system. At least when I tested it out. I was thinking this might be a good candidate in case the Video player was encapsulated. The controls really need to understand the size of the component rather than the size of the page.

Reply:
👍 thanks for clarifying

can we change the class to correspond to the container it's applying against? 

the main class is `VideoPlayer__container` so can we call this one by its modifier `VideoPlayer__container--small`. This is to more concisely describe the characteristics without alluding to the underlying implementation detail.

Reply:
Updated in the following SHA: https://github.com/primer/brand/pull/373/commits/421806a546e1941ec949334a3422e4f70e28faf0